runconfig/hostconfig_solaris.go

```diff
@@ -45,3 +45,17 @@ func ValidateIsolation(hc *container.HostConfig) error {
 func ValidateQoS(hc *container.HostConfig) error {
 	return nil
 }
+
+// ValidateCredentialSpec performs platform specific validation of
+// the credential spec in the hostconfig structure. Solaris does not support
+// this Windows-only feature.
+func ValidateCredentialSpec(hc *container.HostConfig) error {
+	// We may not be passed a host config, such as in the case of docker commit
+	if hc == nil {
```

Comment:
Also not sure if we can have this code only written once in a file for anything not Windows.